tests/test_csc.py

```diff
@@ -34,33 +34,6 @@ async def __aexit__(self, *args):
         await self.csc.close()
 
 
-class NewCscTestCase(salobj.BaseCscTestCase, unittest.IsolatedAsyncioTestCase):
-    def basic_make_csc(self, initial_state, config_dir, simulation_mode):
-        return ATWhiteLightSource.WhiteLightSourceCSC(
-            initial_state=initial_state,
-            config_dir=config_dir,
-            simulation_mode=simulation_mode,
-        )
-
-    '''async def test_state_transitions(self):
-        async with self.make_csc(
-            initial_state=salobj.State.STANDBY,
-            config_dir=TEST_CONFIG_DIR,
-            simulation_mode=1,
-        ):
-            await self.check_standard_state_transitions(
-                enabled_commands=[
-                    "powerLightOn",
-                    "powerLightOff",
-                    "emergencyPowerLightOff",
-                    "setLightPower",
-                    "setChillerTemperature",
-                    "startCooling",
-                    "stopCooling",
-                ]
-            )'''
-
-
 class CscTestCase(asynctest.TestCase):
```

Comment:
Please stop using `asynctest` as it has been abandoned.

Instructions:
* Replace `asynctest.TestCase` with `unittest.IsolatedAsyncioTestCase`
* Replace any existing instances of `async def setUp` and `async def tearDown` with `async def asyncSetUp` and `async def asyncTearDown` (otherwise they will not run).
* Do a global search for `asynctest` and remove all instances (except in version history). Typically there is one in `setup.py`.

Reply:
I strongly suggest updating this test to use `salobj.BaseCscTestCase` instead of a separate `Harness` class. Conversion is easy and it offers many advantages:
* Built-in support for testing standard state transitions
* Built-in support for testing the command-line script (bin/run_ATWhilteLightSource.py)
* Adds `assert_next_summary_state` and `assert_next_sample` methods, which offer a compact way to perform common checks. These can remove a lot of boilerplate.

In fact you've already done most of the work in test_standard_state_transitions.py. I suggest you just move that code here and rewrite these existing tests to use `async with self.make_csc(...):` instead of `async with Harness(...) as harness:`
Then replace all instances of `harness` with `self`, e.g. `harness.remote` -> `self.remote` and `harness.csc` -> `self.csc`.

Reply:
Please add a test of the bin script. It's basically a one-liner with `BaseCscTestCase`